src/OrbitGl/Batcher.cpp

```diff
@@ -40,20 +42,20 @@ void Batcher::AddVerticalLine(Vec2 pos, float size, float z, const Color& color,
   AddLineInternal(pos, pos + Vec2(0, size), z, color, picking_color, nullptr);
 }
 
-void Batcher::AddBox(const Box& box, const std::array<Color, 4>& colors,
+void Batcher::AddBox(const Tetragon& box, const std::array<Color, 4>& colors,
                      std::unique_ptr<PickingUserData> user_data) {
   Color picking_color = PickingId::ToColor(PickingType::kBox, user_data_.size(), batcher_id_);
   AddBoxInternal(box, colors, picking_color, std::move(user_data));
 }
 
-void Batcher::AddBox(const Box& box, const Color& color,
+void Batcher::AddBox(const Tetragon& box, const Color& color,
                      std::unique_ptr<PickingUserData> user_data) {
   std::array<Color, 4> colors;
   colors.fill(color);
   AddBox(box, colors, std::move(user_data));
 }
 
-void Batcher::AddBox(const Box& box, const Color& color, std::shared_ptr<Pickable> pickable) {
+void Batcher::AddBox(const Tetragon& box, const Color& color, std::shared_ptr<Pickable> pickable) {
```

Comment:
The functions should probably also be renamed

Reply:
And @karupayun should have a look how this clashes with his current refactorings

Reply:
Well actually, there even no need for the whole function anymore. AddTrapezium can handle all cases from its logic. I think AddTrapezium should be renamed to AddTetragon and add Box should be removed. OR, we keep both (Trapezium and Box), forward AddBox to AddTrapezium (to get rid of AddBoxInternal) and change AddBox to not take a Tetragon but a pos/size/z pair and calls MakeBox internally.

Reply:
Alternatively, IFF AddBoxInternal is for some reasons faster than 2xAddTriangeInternal, than rename AddBoxInternal to AddTetragonInternal and forward the AddTrapezium call to this rather than to 2xAddTriangle.

Reply:
I am not sure about renaming it. The method might expect the Tetragon to be a box semantically.

Reply:
> I am not sure about renaming it. The method might expect the Tetragon to be a box semantically.

Then we should enforce that with the type system - not with a name.

Reply:
> enforce that with the type system
But if we do, the newly created class (e.g. Box or Rectangle) will have to check whether its vertices actually form a rectangle (the box we used to have didn't do anything like that).

Reply:
There are multiple options:

1. `Box` or `Rectangle` could store `pos`, `size`, and `z`. This would enforce it being a rectangle.
2. `Box` or `Rectangle` could be a class instead of a `struct` and we enforce it being a ractangle by class-invariants.

Reply:
We've agreed with Pablo to get rid of AddTrapezium method in a subsequent PR. AddBox seems to be capable of drawing any tetragon. TODO added.